examples/src/fofb_processing.c

```diff
@@ -1,272 +1,298 @@
-    halcs_number = DFLT_HALCS_NUMBER;
-  }
-  else {
-    halcs_number = strtoul (halcs_number_str, NULL, 10);
+    ret = 1;
+    goto err_free_mem;
+  } else {
+    fprintf(stdout, "[client:fofb_processing]: HALCS client successfully"
+      " created\n");
   }
 
+  // Setting target service name
+  // ########################################################################### 
   char service[50];
-  snprintf (service, sizeof (service), "HALCS%u:DEVIO:FOFB_PROCESSING%u", board_number, halcs_number);
-
-  /* ------------------------------------------------------------------------------ RAM write enable */
-
-  uint32_t ram_write_en;
-  if (ram_write_en_str == NULL) {
-    fprintf (stderr, "[client:fofb_processing]: ram_write_en >> option not recognized or missing argument\n");
-            print_help (argv [0]);
-            exit (1);
-  }
-  else {
-    ram_write_en = strtoul (ram_write_en_str, NULL, 10);
-    if (ram_write_en == 1) {
-      fprintf(stdout, "[client:fofb_processing]: WRITE mode\n");
-    }
-    else if (ram_write_en == 0) {
-      fprintf(stdout, "[client:fofb_processing]: READ mode\n");
+  snprintf(service, sizeof(service), "HALCS%u:DEVIO:FOFB_PROCESSING%u",
+    nec_args.board_slot, nec_args.halcs_number);
+  fprintf(stdout, "[client:fofb_processing]: Target service name: %s\n",
+    service);
+
+  // Performing operation
+  // ########################################################################### 
+  if(nec_args.op == SET_COEFFS) {
+    halcs_client_err_e err;
+
+    err = halcs_fofb_processing_coeff_ram_bank_write(client, service,
+      nec_args.channel, coeffs);
+    if(err != HALCS_CLIENT_SUCCESS) {
+      fprintf(stderr, "[client:fofb_processing]: "
+        "halcs_fofb_processing_coeff_ram_bank_write failed\n");
+
+      ret = 1;
+      goto err_halcs_exit;
+    } else {
+      fprintf(stdout, "[client:fofb_processing]: "
+        "halcs_fofb_processing_coeff_ram_bank_write succeed\n");
     }
-    else {
-      fprintf(stdout, "[client:fofb_processing]: ram write enable value must be 0 or 1\n");
-      print_help (argv [0]);
-      exit (1);
+  } else if(nec_args.op == GET_COEFFS) {
+    halcs_client_err_e err;
+
+    err = halcs_fofb_processing_coeff_ram_bank_read(client, service,
+      nec_args.channel, &coeffs);
+    if(err != HALCS_CLIENT_SUCCESS) {
+      fprintf(stderr, "[client:fofb_processing]: "
+        "halcs_fofb_processing_coeff_ram_bank_read failed\n");
+
+      ret = 1;
+      goto err_halcs_exit;
+    } else {
+      fprintf(stdout, "[client:fofb_processing]: "
+        "halcs_fofb_processing_coeff_ram_bank_read succeed\n");
+
+      fprintf(stdout, "{");
+      for(int i = 0;
+        i < FOFB_PROCESSING_REGS_RAM_BANK_SIZE/sizeof(uint32_t); i++) {
+          if(i != (FOFB_PROCESSING_REGS_RAM_BANK_SIZE/sizeof(uint32_t) - 1)) {
+            fprintf(stdout, "%d, ", coeffs.data[i]);
+          } else {
+            fprintf(stdout, "%d", coeffs.data[i]);
+          }
+        }
```

Comment:
```suggestion
        i < FOFB_PROCESSING_REGS_RAM_BANK_SIZE/sizeof(uint32_t) - 1; i++) {
        fprintf(stdout, "%d, ", coeffs.data[i]);
      }
      fprintf(stdout, "%d", coeffs.data[i]);
```

Me parece um pouco mais limpo? Aí declara `int i` fora do for para funcionar.

Reply:
Um detalhe que não sei se importa é que imprimir com  `GET_COEFF` não gera um arquivo que vc possa ler de volta depois, né?

Reply:
Sim, de fato. Tal implementação ajudaria a scriptar um teste de escrita/leitura fazendo diff dos arquivos, né?
Vou deixar essa conversa em aberto e futuramente implemento.

Reply:
Implementado!